src/main/java/com/synopsys/integration/detect/lifecycle/run/step/IntelligentModeStepRunner.java

```diff
@@ -47,15 +49,17 @@ public class IntelligentModeStepRunner {
     private final Logger logger = LoggerFactory.getLogger(this.getClass());
     private final StepHelper stepHelper;
     private final Gson gson;
+    private final DetectConfigurationFactory configurationFactory;
 
-    public IntelligentModeStepRunner(OperationRunner operationRunner, StepHelper stepHelper, Gson gson) {
+    public IntelligentModeStepRunner(OperationRunner operationRunner, StepHelper stepHelper, Gson gson, DetectConfigurationFactory configurationFactory) {
```

Comment:
Since configurationFactory.componentLocationAnalysisEnabled() alone is referred in this class, the boolean can be passed to this constructor.